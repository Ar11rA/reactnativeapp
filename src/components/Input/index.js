import React from 'react';
import { TextInput, View, Text } from 'react-native';
import inputStyles from './Input.style'

const Input = (props) => {
  const { textInputStyle, labelStyle, containerStyle } = inputStyles
  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>{props.label}</Text>
      <TextInput secureTextEntry={props.password} placeholder={props.placeholder} autoCorrect={false} style={textInputStyle} value={props.value} onChangeText={ () => props.onChangeText() }/>
    </View>
  );
}

export default Input