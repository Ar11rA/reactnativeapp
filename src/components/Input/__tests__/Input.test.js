import { View, Text, TextInput } from 'react-native';
import React from 'react';
import Input from '../';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import EnzymeAdapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new EnzymeAdapter() });
it('renders correctly', () => {
  const tree = renderer.create(
    <Input />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('it should render 1 view component', () => {
  const wrapper = shallow(<Input/>);
  expect(wrapper.find(View)).toHaveLength(1);
});
  
it('it should render 1 text component', () => {
  const wrapper = shallow(<Input/>);
  expect(wrapper.find(Text)).toHaveLength(1);
});

  
it('it should render 1 text input component', () => {
  const wrapper = shallow(<Input/>);
  expect(wrapper.find(Text)).toHaveLength(1);
});

