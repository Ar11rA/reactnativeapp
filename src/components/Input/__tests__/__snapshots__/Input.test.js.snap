// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
        "paddingLeft": 20,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder={undefined}
    secureTextEntry={undefined}
    style={
      Object {
        "color": "#000",
        "flex": 2,
        "fontSize": 18,
        "lineHeight": 25,
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
    value={undefined}
  />
</View>
`;
